src/witan/app/handler.clj

```diff
@@ -17,32 +17,49 @@
             [witan.app.schema :as w]
             [compojure.api.sweet :as sweet]
             [ring.util.http-response :refer :all]
-            [clojure.tools.logging :as log])
+            [clojure.tools.logging :as log]
+            [clj-time.core :as t]
+            [overtone.at-at :as at])
   (:gen-class))
 
 ;; Global storage for store generated tokens.
 (defonce tokens (atom {}))
 
+;; The at-at thread pool
+(defonce at-at-pool (at/mk-pool))
+
 ;; Authenticate Handler
 ;; Respons to post requests in same url as login and is responsible to
 ;; identify the incoming credentials and set the appropiate authenticated
 ;; user into session. `authdata` will be used as source of valid users.
 
+(defn clear-expired-tokens!
+  [tokens]
+  (->> @tokens
+       (filter (fn [[k {:keys [user expires]}]] (t/before? (t/now) expires)))
+       (into {})
+       (reset! tokens)))
+
+(defn add-token!
+  [user-id tokens]
+  (let [token (user/random-token)
+        ttl (t/days 28)]
+    (swap! tokens assoc (keyword token) {:user user-id :expires (t/plus (t/now) ttl)})
```

Comment:
(pure nitpick) I would usually save the time of creation, because if we modify the ttl it's still possible to calculate whether the token has been expired, but I'm happy with this since it's unlikely to be a problem for now.